GoRogue/MapGeneration/RegionWithComponents.cs

```diff
@@ -0,0 +1,115 @@
+using System.Collections;
+using System.Collections.Generic;
+using GoRogue.Components;
+using GoRogue.Components.ParentAware;
+using SadRogue.Primitives;
+
+namespace GoRogue.MapGeneration
+{
+    /// <summary>
+    /// A Class which contains a Region, and a Component Collection.
+    /// </summary>
+    public class RegionWithComponents : IObjectWithComponents, IReadOnlyArea
+    {
+        /// <summary>
+        /// The region containing the components
+        /// </summary>
+        public Region Region { get; private set; }
```

Comment:
Does the setter being `private` limit the capability to implement custom transformations, etc?  Is there any harm in it being a public field like so?

```suggestion
        public Region Region;
```